components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,159 @@
+    };
+    aes::aes_gcm_decrypt(jwe, &EncryptionAlgorithm::A256GCM, &secret)
+}
+
+#[test]
+fn test_simple_roundtrip() {
+    // We should be able to round-trip data.
+    use super::{decrypt_jwe, encrypt_to_jwe, DecryptionParameters, EncryptionParameters};
+    let jwk = Jwk::new_symmetric_key(Some("my key".to_string())).unwrap();
+    let data = "to be, or not 🐝🐝";
+    let encrypted = encrypt_to_jwe(
+        data.as_bytes(),
+        EncryptionParameters::Direct {
+            jwk: &jwk,
+            enc: EncryptionAlgorithm::A256GCM,
+        },
+    )
+    .unwrap();
+    let decrypted = decrypt_jwe(&encrypted, DecryptionParameters::Direct { jwk }).unwrap();
+    assert_eq!(data, decrypted);
+}
+
+#[test]
+fn test_iv() {
+    // Encrypting the same thing twice should give different payloads due to
+    // different IV.
+    use super::{encrypt_to_jwe, EncryptionParameters};
+    let jwk = Jwk::new_symmetric_key(Some("my key".to_string())).unwrap();
+    let data = "to be, or not 🐝🐝";
+    let e1 = encrypt_to_jwe(
+        data.as_bytes(),
+        EncryptionParameters::Direct {
+            enc: EncryptionAlgorithm::A256GCM,
+            jwk: &jwk,
+        },
+    )
+    .unwrap();
+    let e2 = encrypt_to_jwe(
+        data.as_bytes(),
+        EncryptionParameters::Direct {
+            jwk: &jwk,
+            enc: EncryptionAlgorithm::A256GCM,
+        },
+    )
+    .unwrap();
+    assert_ne!(e1, e2);
+}
+
+#[test]
+fn test_jose() {
+    // ciphertext generated by node-jose via:
+    /*
+    const parseJwk = require("jose/jwk/parse").default;
+    const CompactEncrypt = require("jose/jwe/compact/encrypt").default;
+    const encoder = new TextEncoder();
+    const payload = "Hello, World!";
+    const key = "asecret256bitkeyasecret256bitkey";
+    parseJwk({kty: "oct", k: Buffer.from(key).toString("base64")}, "A256GCM").then(key => {
+        new CompactEncrypt(encoder.encode(payload))
+            .setProtectedHeader({ alg: "dir", enc: "A256GCM" })
+            .encrypt(key)
+            .then(jwe => {
+                console.log(jwe);
+            });
+    })
+    */
+    // (A note for future readers - we tried using python-jose, but it
+    // generated a 16 byte nonce, where the spec clearly calls for exactly 12
+    // bytes. We could decrypt that python-jose payload if we modified
+    // `Nonce::try_assume_unique_for_key()` to allow a longer key, but we don't
+    // want to do that until we have evidence it's actually spec compliant.)
+    use super::{decrypt_jwe, DecryptionParameters};
+    let jwk = Jwk::new_symmetric_from_bytes(None, "asecret256bitkeyasecret256bitkey".as_bytes());
+    let ciphertext = "eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..nhKdQEKqoKPzfCda.rQOj0Nfs6wO5Gj4Quw.CMJFS9YBADLLePdj1sssSg";
+    let decrypted = decrypt_jwe(ciphertext, DecryptionParameters::Direct { jwk }).unwrap();
+    assert_eq!(decrypted, "Hello, World!");
+}
+
+#[test]
+fn test_bad_key_type() {
```

Comment:
It may also be worth adding some tests for things that are supposed to fail, e.g.:

* Try to decrypt with the wrong key
* Try to decrypt after modifying the ciphertext